backend/src/main/kotlin/code/nebula/cipherquest/service/GameService.kt

```diff
@@ -70,6 +72,26 @@ class GameService(
         }
     }
 
+    private fun detectCheat(userToQuery: Pair<UserLevel, String>): BotMessage? {
+        val (userLevel, userMessage) = userToQuery
+        val userId = userLevel.userId
+
+        return if (Regex(winCondition).containsMatchIn(userMessage) &&
```

Comment:
We can use the `takeIf` kotlin operator.
https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/take-if.html